From: Dawei Li <set_pte_at@outlook.com>
Subject: mm/damon: creating kmem cache for damon regions by KMEM_CACHE()
Date: Wed, 14 Sep 2022 00:21:58 +0800

Damon regions are dynamic objects which can be created and destroyed
frequently, a dedicated slab cache is created by KMEM_CACHE(), as
suggested by akpm.

Link: https://lkml.kernel.org/r/Message-ID:
Signed-off-by: Dawei Li <set_pte_at@outlook.com>
Reviewed-by: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/core.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mm/damon/core.c~mm-damon-improve-damon_new_region-strategy-fix
+++ a/mm/damon/core.c
@@ -1289,8 +1289,7 @@ bool damon_find_biggest_system_ram(unsig
 
 static int __init damon_init(void)
 {
-	damon_region_cache = kmem_cache_create("damon_region_cache", sizeof(struct damon_region),
-		0, 0, NULL);
+	damon_region_cache = KMEM_CACHE(damon_region, 0);
 	if (unlikely(!damon_region_cache)) {
 		pr_err("creating damon_region_cache fails\n");
 		return -ENOMEM;
_
